### PR TITLE
interacting with graphql apis in r blogpost

### PR DESCRIPTION
**Pull request by denironyx**
Hi @steffilazerte, 

Here is the pull request for the blogpost. Please, kindly take a look at it and let me know what you think about it.

Thanks!

cc @sckott

### CODE REVIEW

**Review by steffilazerte — COMMENTED** [review]
Hi @denironyx,

Thank you for the fun and engaging post. It is well written (I don't think I've ever found so few typos!), and the only major suggestions I have is that 1) you add some connecting sentences to help the flow of the article (see my direct comments), and 2) you consider rearranging the first couple of sections: 

- Start with "So what is GraphQL?" and include a bit why you wanted to use it
- "And then, there is R", include why you wanted to access GraphQL from R
- "Moving on" consider renaming this to "Accessing GraphQL from R"
- Finally "How did I find out about ghql?"

This way the article starts out general and narrows in on the package of interest. And with the little details added along the way, it becomes more of a story.

I have about 20 comments made directly on the Rmd file in GitHub (although not all require action from you). Take a look and see how you want to address them.

Let me know how it goes!

* [X] post follows Content Guidelines
* [X] post follows Style Guide
* [X] title is in Title Case
* [ ] publication date is ok
  - **We'll figure this out later**
* [X] alternative text of images is informative
* [ ] Twitter metadata looks ok (paste post preview link in [Twitter card validator](https://cards-dev.twitter.com/validator); might have to click twice on "Preview card")
* [X] author metadata is provided with correct folder name
* [X] html not included in pull request of Rmd post
* [X] I ran `roblog::ro_lint_md()` on index.md
* [X] I ran `roblog::ro_check_urls()` on index.md
* [X] I ran a spell-check on index.md
* [ ] YAML subject tags are ok

File: content/blog/2020-10-10-interacting-with-grahpql-apis-in-r/index.Rmd [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+## And then, there is R
+R is a free software environment for statistical computing and graphics. Well, that's how it's officially defined. But, trust me its gradually evolving to do more than statistical computing and making of beautiful graphics.
+
+## Moving on
+In order for R to interact with any GraphQL API, it requires a GraphQL client. That's where `ghql`, a GraphQL client for R, developed by Scott Chamberlain comes into play. Still confused? I hope the flow chart below helps you make more sense of it.
+{{< figure src = "GHQL.png" width = "350" alt = "GraphQL client and R connection flow" caption = "GraphQL client and R connection flow" class = "center">}}
+
+## Working with `Countries List`, a GraphQL public API
+The [Countries GraphQL API](https://github.com/trevorblades/countries) is a public GraphQL API for information about countries, continents, and languages. This public API uses [Countries List](https://annexare.github.io/Countries/) and [provinces](https://github.com/substack/provinces) as data sources, so the schema follows the shape of that data, with a few exceptions:
```
[the remark this answers is not in the document]
Comment (stefaniebutland) [reply content/blog/2020-10-10-interacting-with-grahpql-apis-in-r/index.Rmd:39]: I agree it's important to add some text to explain the examples.

**Comment by stefaniebutland** [comment]
Hi @denironyx. I agree with @steffilazerte that readers will benefit from a few explanatory sentences to help them understand more of your thinking. Looking forward to seeing this on the rOpenSci blog :-)

File: content/blog/2020-10-10-interacting-with-grahpql-apis-in-r/index.Rmd [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+---
+
+## How did I find out about ghql?
+After spending days researching about the R packages that can interact with a GraphQL API, I found three packages. I picked `ghql` over the others because, it was an rOpenSci package. 
+
+## So what is GraphQL?
+GraphQL is a query language for application programming interfaces (APIs) that prioritizes giving clients exactly the data they request. It's designed to make APIs flexible, fast and friendly. Basically, it is used to load data from a server to a client and it does this in a much more efficient manner than traditional methods and services.
+
+## And then, there is R
+R is a free software environment for statistical computing and graphics. Well, that's how it's officially defined. But, trust me its gradually evolving to do more than statistical computing and making of beautiful graphics.
```
[the remark this answers is not in the document]
Comment (denironyx) [reply content/blog/2020-10-10-interacting-with-grahpql-apis-in-r/index.Rmd:32]: Thanks

**Review by steffilazerte — COMMENTED** [review]
Hi @denironyx,

Thank you for your time and effort! I absolutely love your introduction, it's the kind of story that I think a lot of readers will identify with and it's written in a way that is fun and easy to read.

I have 9 final suggestions, 4 relate to moving the text from "What is GraphQL" and "And then, there is R" into the first two sections, 1 suggests wording for a final, concluding sentence, 1 for the publication date, and the rest are just a couple of wording suggestions. 

@stefaniebutland has given us the publication date of Tuesday December 8th, would you be able to make these last changes by **Friday December 4th**? If so, we'll aim for publishing on the 8th!

Thanks again, 

* [X] post follows Content Guidelines
* [X] post follows Style Guide
* [X] title is in Title Case
* [ ] publication date is ok - **Update all dates to 2020-12-08**
* [X] alternative text of images is informative
* [X] Twitter metadata looks ok (paste post preview link in [Twitter card validator](https://cards-dev.twitter.com/validator); might have to click twice on "Preview card")
* [X] author metadata is provided with correct folder name
* [X] html not included in pull request of Rmd post
* [X] I ran `roblog::ro_lint_md()` on index.md
* [X] I ran `roblog::ro_check_urls()` on index.md
* [X] I ran a spell-check on index.md
* [X] YAML subject tags are ok

**Comment by steffilazerte** [comment]
Awesome! I made two small changes (I think you've worked hard enough!) 1) I removed two instances of a paragraph that got triplicated, 2) I changed the date of the folder to the publication date.

We're ready for publication next Tuesday, thank you so much for your time and effort on this blog post. Your work is done :) I'll post a draft tweet here for how we'll advertise your blog post, which you're welcome to comment on, but if you're happy with it, no need to bother!

**Comment by steffilazerte** [comment]
Draft tweet:

---

[blog] Accessing GrahpQL from R

@denironyx considers tricky data sources and demonstrates how to use the R package ghql to access GraphQL APIs.

https://ropensci.org/blog/2020/12/08/accessing-grahpql-in-r/

GHQL
[ALT TEXT]: Diagram showing GraphQL client and R connection flow

#rstats #GraphQL #ghql

**Comment by stefaniebutland** [comment]
@steffilazerte please tag Scott in the tweet since he's co-author.

**Comment by stefaniebutland** [comment]
@denironyx Dennis, your post is a great read. Proud that we're able to share this from rOpenSci

**Comment by denironyx** [comment]
Hurray! Thanks, @stefaniebutland for your contribution. A huge thank you to the amazing @steffilazerte!

**Pull request merged**